### PR TITLE
PLANET-7584 Fix PostsList block in the editor

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7584](https://jira.greenpeace.org/browse/PLANET-7584)

In this PR I also create separate files for the ActionsList/PostsList block editor styles, to be more consistent with other blocks and for more clarity.

**Related PR:** https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1238

### Testing

Make sure that both the Actions List and Posts List block look good in the editor and also in the frontend, whether the plugin is activated or not. You can use [this page](https://www-dev.greenpeace.org/test-jupiter/the-lists/) for testing if you'd like.

Before these changes this is what the broken version of the Posts List block looked like: 

bug

### CODE REVIEW

**Review by mardelnet — COMMENTED** [review]
@mleray  
With the changes implemented in this PR and the plugin active, some elements look odd in the editor:
Screenshot from 2024-09-11 12-09-55
Screenshot from 2024-09-11 12-09-09

**Comment by mardelnet** [comment]
> @mleray With the changes implemented in this PR and the plugin active, some elements look odd in the editor: 

Forget my previous comment, @mleray 
I forgot to switch to [this branch](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/tree/editor-layout-fix) simultaneously.
Sorry for the misunderstanding.

File: assets/src/scss/blocks/PostsList/PostsListEditor.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.posts-list {
+  .carousel-controls {
```
Comment (mardelnet) [inline assets/src/scss/blocks/PostsList/PostsListEditor.scss:2]: I think we can move this class within `&.is-custom-layout-list` and get rid of the `!important` statement.

File: assets/src/scss/editorOverrides.scss [hunk 2]
```diff
@@ -399,57 +399,6 @@ input.describe[type=text][data-setting=caption] {
...
-
-  .is-layout-flow > .is-layout-flow {
-    margin-top: -36px;
-  }
-
-  @media (max-width: 991px) {
-    .wp-block-post-title,
-    .wp-block-post-excerpt {
-      margin-top: -12px;
-    }
```
Comment (mardelnet) [inline assets/src/scss/editorOverrides.scss:445]: Have we moved these classes to the new SCSS files?

File: assets/src/scss/editorOverrides.scss [hunk 3]
```diff
@@ -399,57 +399,6 @@ input.describe[type=text][data-setting=caption] {
...
-  .wp-block-group.is-layout-flow {
-    flex-basis: 65%;
-  }
-
-  .wp-block-post-title {
-    margin-top: -25px !important;
-  }
-
-  .wp-block-post-excerpt {
-    margin-top: -24px;
```
Comment (mardelnet) [inline assets/src/scss/editorOverrides.scss:430]: Have we moved these classes to the new SCSS files?

File: assets/src/scss/blocks.scss [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // Beta blocks
-@import "blocks/ActionsList";
-@import "blocks/PostsList";
+@import "blocks/ActionsList/ActionsList";
+@import "blocks/PostsList/PostsList";
```
Comment (mardelnet) [inline assets/src/scss/blocks.scss:3]: For consistency, these files should be named `ActionsListStyle` and `PostsListStyle`

File: assets/src/scss/editorStyle.scss [hunk 5]
```diff
@@ -45,6 +45,8 @@
 @import "blocks/Submenu/SubmenuEditorStyle";
 @import "blocks/TakeActionBoxout/TakeActionBoxoutEditorStyle";
 @import "blocks/Timeline/TimelineEditorStyle";
+@import "blocks/PostsList/PostsListEditor";
+@import "blocks/ActionsList/ActionsListEditor";
```
Comment (mardelnet) [inline assets/src/scss/editorStyle.scss:49]: For consistency, these files should be named `ActionsListEditorStyle` and `PostsListEditorStyle`

**Review by mardelnet — COMMENTED** [review]
I think the "Add read more..." button is no longer visible:

 Edit-Post-“Duis-posuere-6”-‹-Greenpeace-—-WordPress

**Comment by mleray** [comment]
> I think the "Add read more..." button is no longer visible:

@mardelnet that's [on purpose](https://github.com/greenpeace/planet4-master-theme/pull/2374/files#diff-810f3000c35003ed63fb41d87b35e7a878f914391e50a52a54fd102ec1d20fabR2-R4), we actually don't want this block to show (same as the Posts List)

File: assets/src/scss/editorOverrides.scss (hunk 2 above)
Comment (mleray) [reply assets/src/scss/editorOverrides.scss:445]: No, these are no longer needed if we remove the custom margins from the plugin

File: assets/src/scss/blocks.scss (hunk 4 above)
Comment (mleray) [reply assets/src/scss/blocks.scss]: Updated!

File: assets/src/scss/editorStyle.scss (hunk 5 above)
Comment (mleray) [reply assets/src/scss/editorStyle.scss]: Updated!

File: assets/src/scss/blocks/PostsList/PostsListEditor.scss (hunk 1 above)
Comment (mleray) [reply assets/src/scss/blocks/PostsList/PostsListEditor.scss]: In the editor we actually always want to hide the carousel controls, not only for the list style

**Review by mardelnet — APPROVED** [review]
LGTM!

**Review by mardelnet — APPROVED** [review]
Good job!
BTW, do you know what is it used for the hook on line 407 of the `functions.php` file which also overwrites styles of the `block-editor-block-list__layout` class?

**Comment by mleray** [comment]
> block-editor-block-list__layout

@mardelnet it seems to be for the feature Nikos showed us during the sprint meeting, for the News & Stories page to be disabled in the editor

**Pull request merged**